### PR TITLE
Multi shard DELETE with LIMIT clause should not be supported

### PR DESCRIPTION
**Pull request by rafael**
### Desc

Per discussion, this was too radical. We should not support this.

### CODE REVIEW

File: go/vt/vtgate/planbuilder/dml.go [hunk 1]
```diff
@@ -185,8 +185,14 @@ func buildDeletePlan(del *sqlparser.Delete, vschema VSchema) (*engine.Route, err
...
 	}
-	if len(er.Table.Owned) != 0 && er.Opcode == engine.DeleteSharded {
-		return er, errors.New("unsupported: multi shard delete on a table with owned lookup vindexes")
+
+	if er.Opcode == engine.DeleteSharded {
+		if len(er.Table.Owned) != 0 {
+			return er, errors.New("unsupported: multi shard delete on a table with owned lookup vindexes")
+		}
+		if del.Limit != nil {
+			return er, errors.New("unsupported: multi shard delete limit clause in DML")
```
Comment (demmer) [inline go/vt/vtgate/planbuilder/dml.go]: I think a better error message would be "unsupported: multi shard delete with limit"

The "DML" part is implicit

Comment (rafael) [reply go/vt/vtgate/planbuilder/dml.go]: Sounds good. I added that following the pattern by existent error messages.

**Comment by sougou** [comment]
LGTM

**Comment by alainjobart** [comment]
@demmer I think your comment has been addressed, if you agree I'll merge.

**Pull request merged**